src/js/media-time-range.js

```diff
@@ -313,77 +327,84 @@ class MediaTimeRange extends MediaChromeRange {
-    };
+    this.updatePointerBar(evt);
 
-    // Trigger when the mouse moves over the range
-    let rangeEntered = false;
-    let rangepointermoveHander = () => {
-      const mediaDurationStr = this.getAttribute(
-        MediaUIAttributes.MEDIA_DURATION
-      );
-      if (!rangeEntered && mediaDurationStr) {
-        rangeEntered = true;
-        trackMouse();
-
-        let offRangeHandler = (evt) => {
-          if (
-            !evt.composedPath().includes(this) ||
-            [...boxes].some((b) => evt.composedPath().includes(b))
-          ) {
-            window.removeEventListener('pointermove', offRangeHandler);
-            rangeEntered = false;
-            stopTrackingMouse();
-          }
-        };
-        window.addEventListener('pointermove', offRangeHandler, false);
-      }
-    };
-    this.addEventListener('pointermove', rangepointermoveHander, false);
+    const duration = +this.getAttribute(MediaUIAttributes.MEDIA_DURATION);
+    // If no duration we can't calculate which time to show
+    if (!duration) return;
+
+    // Get mouse position percent
+    const rangeRect = this.range.getBoundingClientRect();
+    let mousePercent = (evt.clientX - rangeRect.left) / rangeRect.width;
+    // Lock between 0 and 1
+    mousePercent = Math.max(0, Math.min(1, mousePercent));
+
+    const boxPos = getBoxPosition(this, this.#previewBox, mousePercent);
+    const { style } = getOrInsertCSSRule(
+      this.shadowRoot,
+      '[part~="preview-box"]'
+    );
+    style.transform = `translateX(${boxPos}px)`;
+
+    const detail = mousePercent * duration;
+    const mediaPreviewEvt = new window.CustomEvent(
+      MediaUIEvents.MEDIA_PREVIEW_REQUEST,
+      { composed: true, bubbles: true, detail }
+    );
+    this.dispatchEvent(mediaPreviewEvt);
+  }
+
+  // Trigger when the mouse moves over the range
+  #rangeEntered = false
+
+  #offRangeHandler = (evt) => {
+    if (
+      !evt.composedPath().includes(this) ||
+      [...this.#boxes].some((b) => evt.composedPath().includes(b))
+    ) {
+      window.removeEventListener('pointermove', this.#offRangeHandler);
+      this.#rangeEntered = false;
+      this.#stopTrackingMouse();
+    }
+  }
+
+  #trackMouse = () => {
+    window.addEventListener('pointermove', this.#pointermoveHandler, false);
+  }
+
+  #stopTrackingMouse = () => {
+    window.removeEventListener('pointermove', this.#pointermoveHandler);
+    const endEvt = new window.CustomEvent(
+      MediaUIEvents.MEDIA_PREVIEW_REQUEST,
+      { composed: true, bubbles: true, detail: null }
+    );
+    this.dispatchEvent(endEvt);
+  }
+
+  #rangepointermoveHandler = () => {
+    const mediaDurationStr = this.getAttribute(
+      MediaUIAttributes.MEDIA_DURATION
+    );
+    if (!this.#rangeEntered && mediaDurationStr) {
+      this.#rangeEntered = true;
+      this.#trackMouse();
+
+      window.addEventListener('pointermove', this.#offRangeHandler, false);
+    }
+  }
+
+  _enableBoxes() {
```

Comment:
feel free to change these to private class methods too.
at the time I didn't know we were adopting these in Media Chrome.

Reply:
I guess worth input from others. I'm ok with either, since esbuild does compile them away.

Reply:
The other private props are more important to have the event handler methods be per instance to avoid wardresses around removing the event handler and all buttons stopping working.

Reply:
it's fine I think, support is pretty good https://caniuse.com/mdn-javascript_classes_private_class_fields

Reply:
yeah, we could consider having esbuild not copmile it away.